File: src/main/java/de/dk/api/codeChallenge/notification/model/dto/CreateTypeRequest.java
package de.dk.api.codeChallenge.notification.model.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;

public record CreateTypeRequest(
        @NotBlank(message = "name must not be blank")
        String name,

        @NotBlank(message = "category must not be blank")
        @Pattern(regexp = "A|B", message = "category must be either A or B")
        String category
) {
}

File: src/main/kotlin/de/dkb/api/codeChallenge/notification/model/dto/CreateTypeRequest.kt
package de.dkb.api.codeChallenge.notification.model.dto

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Pattern
data class CreateTypeRequest(

    @field:NotNull
    val name: String,
    @field:Pattern(regexp = "A|B")
    val category: String
)